plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointEntryIterator.java

```diff
@@ -458,8 +459,13 @@ private DeltaLakeTransactionLogEntry buildAddEntry(ConnectorSession session, Blo
         log.debug("Block %s has %s fields", block, addEntryRow.getFieldCount());
         CheckpointFieldReader add = new CheckpointFieldReader(session, addEntryRow, type);
 
```

Comment:
Since the input `block` to already a loaded block and we call `block.getObject(pagePosition, SqlRow.class)`, I think we will read and decode all the fields in ADD struct even if partition pruning takes place.
Ideally we should process only the partition values fields till the `partitionMatchesPredicate` check to get benefit of lazy loading within ADD struct.